test/jdk/tools/jar/ManifestDirectoryCompression.java

```diff
@@ -48,38 +51,43 @@ public class ManifestDirectoryCompression {
             ToolProvider.findFirst("jar")
                     .orElseThrow(() -> new RuntimeException("jar tool not found"));
 
+    private Path tempDir;
+
+    @BeforeMethod
+    private void setUp() throws Exception {
+        tempDir = Files.createTempDirectory("temp");
+    }
+
     /** Remove dirs & files needed for test. */
-    private static void cleanup(Path dir) {
+    @AfterMethod
+    private void cleanup() {
+        deleteRecursively(tempDir);
+    }
+
+    private static void deleteRecursively(Path path) {
         try {
-            if (Files.isDirectory(dir)) {
-                try (Stream<Path> s = Files.list(dir)) {
-                    s.forEach(p -> cleanup(p));
+            if (Files.isDirectory(path)) {
+                try (Stream<Path> s = Files.list(path)) {
+                    s.forEach(p -> deleteRecursively(p));
                 }
             }
-            Files.delete(dir);
+            Files.delete(path);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
     @Test
-    public void run() throws Throwable {
-        Path topDir = Files.createTempDirectory("delete");
-        try {
-            Path entry = Files.writeString(topDir.resolve("test.txt"), "Some text...");
-            doTest(topDir.resolve("test.jar"), entry);
-        } finally {
-            cleanup(topDir);
-        }
-    }
-
-    private static void doTest(Path jar, Path entry) throws Throwable {
-        String[] jarArgs = new String[] {"cf", jar.toString(), entry.toString()};
+    public void run() throws Exception {
+        Path entryPath = Files.writeString(tempDir.resolve("test.txt"), "Some text...");
+        Path jar = tempDir.resolve("test.jar");
```

Comment:
Please see comment above regarding the cleanup method.

One other thought you could consider given you only create a jar and file to add to the jar, is to simply add File.deleteIfExists() calls and not bother with a cleanup method given the test case is small and pretty straight forward.  Your choice though :-)

Reply:
Thanks - I left the cleanup in an `@After` method for now, but I'm happy to change it.

One small reason to keep the `@After` method is that I think it ensures the cleanup happens even if the test fails and exits early, without having to wrap the test body in a `try`/`finally`.